pkg/hubble/parser/threefour/parser.go

```diff
@@ -174,10 +174,16 @@ func (p *Parser) Decode(data []byte, decoded *pb.Flow) error {
 	}
 
 	ether, ip, l4, srcIP, dstIP, srcPort, dstPort, summary := decodeLayers(p.packet)
-	if tn != nil && !tn.OriginalIP().IsUnspecified() {
-		srcIP = tn.OriginalIP()
+	if tn != nil {
+		if !tn.OriginalIP().IsUnspecified() {
+			srcIP = tn.OriginalIP()
+			if ip != nil {
+				ip.Source = srcIP.String()
+			}
+		}
+
 		if ip != nil {
-			ip.Source = srcIP.String()
+			ip.Encrypted = (tn.Reason & monitor.TraceReasonEncryptMask) != 0
```

Comment:
Something to keep in mind here is that not all trace points set `tn.Reason`. So we need to be careful not confuse users with false negatives (we had a similar issue with the "is_reply" field).

A quick grep in the bpf sources seems to indicate that only the `from-network` and `from-overlay` trace points set `TRACE_REASON_ENCRYPTED`. I have not tested it, but does this mean that notably `to-network` will always report `Encrypted` as false?

Is this a problem? If so, we should either try to address that in the datapath.

Reply:
Good point. It's something we need to fix one layer below though. We need to be consistent. It must be possible to trust the flag.